azurelinuxagent/common/protocol/wire.py

```diff
@@ -108,6 +108,22 @@ def update_host_plugin_from_goal_state(self):
     def get_endpoint(self):
         return self.client.get_endpoint()
 
+    def init_sysinfo(self, imds_client):
+        vminfo = VMInfo()
```

Comment:
````
vminfo = VMInfo()
compute_info = ComputeInfo()
````
These constructors don't call any other methods and can't throw. My intention was to pass an empty object (with member variables set to `None`, but the references still exist) to the `sysinfo.set_vminfo(vminfo, compute_info` call, since then the references to the (empty) attributes wouldn't throw.

Reply:
ok

Reply:
so if we fail to create a valid object all the values will be None? we should try to choose some appropriate default since code using VMInfo may choke on None.